packages/flutter_tools/lib/src/globals.dart

```diff
@@ -66,13 +66,11 @@ FileSystem get fs => ErrorHandlingFileSystem(
   platform: platform,
 );
 
-final FileSystemUtils _defaultFileSystemUtils = FileSystemUtils(
+FileSystemUtils get fsUtils => context.get<FileSystemUtils>() ?? FileSystemUtils(
```

Comment:
This is why we would sometimes use the real fs

Reply:
Can we make `FileSystemUtils` `const`?

Reply:
ahh, nevermind, then its dependencies would have to be const too...